### ISSUE TITLE
[REQ] ZipFile release of WinPython

### ISSUE THREAD

**Issue by brianddk** [winpython/winpython#633]
Since the installer is only copying files to a specific directory, is there any way of adding a no-installer release of WinPython?  I had hoped to grab the release and install it on an in-execution win-pe instance, but the installer won't run on WinPE.  I then tried to just extract the files from the installer, which worked, but the resulting directory structure ($_OUTDIR) seemed incorrect.

Then again... Could all be vc_redist_x64.exe dependencies (which aren't met in PE either)

**Comment by stonebig** [comment]
in pure truth I don't know what is WinPE, I may have confused with Windows CE

**Comment by stonebig** [comment]
unziping (with 7zip), you shall get directly what you want + a '$PLUGINSDIR' directory to just remove

![unzip_direct](https://user-images.githubusercontent.com/4312421/39766940-ce69037a-52e5-11e8-82b1-899ce95b5e8f.JPG)

**Comment by brianddk** [comment]
[WinPE](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/winpe-intro) is used to bootstrap installs and deployments of Windows.  If you know Linux, the analogy is the Linux-Live-CD.  WinPE is a Windows-Live-CD

[WinRE](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/winpe-intro) is the recovery environment that you get if you ever have a system failure in Windows.

Both WinPE and WinRE use the Windows 10 kernel and hal.  Just like Windows, just different restrictions.

Trying Py2.7 now

**Comment by stonebig** [comment]
ah, knoppix, what a splash it was

**Comment by stonebig** [comment]
sucess ?

**Comment by brianddk** [comment]
Yep... This is all redist / runtime spaghetti (winsxs) stuff I'm running into.  

python27 fails with "side-by-side configuration" problem.

Looks like I just need some system DLLs that are not natively there.  I'll have to run it down when I get more time, but if you guys are ever interested... adding WinPE support would be a big win for sys-admins

**Comment by stonebig** [comment]
will ask for you to @zooba

**Comment by stonebig** [comment]
also "side-by-side" suggests you may just have to remove some dlls in python-2.7x.yg\DLLs

**Comment by stonebig** [comment]
it's all about vsstudio stuff, https://social.msdn.microsoft.com/Forums/vstudio/en-US/7929bbd1-aa7a-42f7-9915-afecf9d35353/sidebyside-configuration-is-incorrect-from-winpe-20?forum=vcgeneral

so the right door for you to knock is the "Python for Visual Studio" folks, or StackOverflow

**Comment by brianddk** [comment]
Yeah... I was following that thread which led me here.  Apparently there was a bug in your manifest maker that caused some of my issues.  I whacked on the manifest file and made it whole again.  If I have some time to kill, I'll submit a PR, but for now I opened a new issue on the broken manifest in release 1.7

https://github.com/winpython/winpython/issues/634

**Comment by stonebig** [comment]
also, it's a nsis wrapper, so you can do 
"WinPython-64bit-2.7.13.1Zero.exe /S /D d:\place_me_there"

see http://nsis.sourceforge.net/Docs/Chapter3.html#usageexamples

**Comment by brianddk** [comment]
Installer has x86 dependencies.

Most x64 WinPE environments don't have WoW64 (ability to run x86 on x64).  The loader kicked the installer out as soon as it saw the PE block stating i386

**Comment by stonebig** [comment]
ok. so I need a PR or some links to figure out what to fix.

Would your requirements be also satisfied  with a Python-3 version ?

**Comment by brianddk** [comment]
Actually.... I just got `WinPython64-3.5.4.2Zero.exe` working in WinPE.  Everything was fine... I just need to install `VC_redist.x64.exe` in the WinPE instance first.  That was tough, but walking through [this](http://www.applepie.se/extract-msi-from-visual-c-2012-redistributable) followed by [this](https://superuser.com/questions/307678/how-do-i-extract-files-from-an-msi-package/307679#307679), followed by [this](http://reboot.pro/topic/4720-visual-c-2005-redistributable-package-x64/?p=57534) basically got me there.

All the problems I saw with 3.5 were msCRT related (which you've abundantly documented) and all the problems with 2.7 were related to issue https://github.com/winpython/winpython/issues/634

Think we're good to go.  One gotcha that stumped me was the difference in older versions of 7-Zip.  With older versions, it names the directory `$_OUTPUT`.  Interestingly enough, Python pukes on paths with special characters like `$` in them.  Likely a CPython upstream bug.  Anyway [7-Zip v18.05](https://sourceforge.net/projects/sevenzip/files/7-Zip/18.05/) used more bland directory names.  Just in case anyone hits this thread looking to do the same.

**Comment by brianddk** [comment]
PS.. If you ever want to try this out for yourself... here's an easy way to run PE.

1. Grab a Windows ISO from the [Evaluation Center](https://www.microsoft.com/en-us/evalcenter/), or grab [HyperV](https://www.microsoft.com/en-us/evalcenter/evaluate-hyper-v-server-2016) if you need more than 90 days
2. Boot the ISO (easy to put on USB key these days)
3. At setup, hit [Shift-F10](https://docs.microsoft.com/en-us/windows-hardware/manufacture/desktop/windows-setup-installing-using-the-mbr-or-gpt-partition-style) to break into a shell.
4. Try to install WinPython, or any other software.

**Comment by stonebig** [comment]
what use would I, or students, have of it ?

**Comment by stonebig** [comment]
answer from @zooba:

````
but if the issue was the CRT, then just copy it out of the Windows SDK redist folder alongside Python (a.k.a. an "app-local install")
````

**Comment by brianddk** [comment]
Yep... Didn't think of scrubbing the WDK iso for the redist files... but
he's right... WAY easier than hacking MSIs.

All sorted now... Thx

**Comment by xaratustrah** [comment]
Hi, sorry to reopen this issue, but running the EXE installer (in my case Winpython-64-3.6.7.0Qt5.exe) gets blocked by the group policy of my Windows 10 machine, so I need to ask an administrator to run the installer each time.

But wouldn't be simpler to just provide a Zip file of winpython instead as the OP suggested?

I also tried to run 7-Zip on the EXE Installer, but it says that it can not open file as archive.

Any hints? thanks.

**Comment by stonebig** [comment]
As an experiment, can you unzip the winpython-3.8.0.0dota2 ? 
It's 7zip with auto extract.

.zip is too big, .7z is too  special, so an auto-extract 7zip is the likely future.
Ideally, with zstd compression, but for that we need standard 7zip to integrate the decompression part.

**Comment by xaratustrah** [comment]
whrere do I get the file "winpython-3.8.0.0dota2" ?

btw. I tried to make a file size comparison:

exe file --> 477 MB
ZIP file --> 799 MB
RAR file --> 648 MB

wouldn't RAR be an alternative?

**Comment by stonebig** [comment]
https://sourceforge.net/projects/winpython/files/WinPython_3.8/3.8.0.0/alphas/

**Comment by xaratustrah** [comment]
this one is good! although running the EXE is still blocked by Win10 policy, but using 7-Zip to extract works perfectly (right-click, 7-zip --> Extract files). Will you be providing future releases like this?

many thanks for the good job with WinPython.

**Comment by stonebig** [comment]
I'll experiment the change for next build cycle. Will need to create a licence file next to the icons.

**Comment by stonebig** [comment]
will now provide .7z release
